.github/workflows/check-semgrep-pro.jsonnet

```diff
@@ -7,53 +7,60 @@ local gha = import 'libs/gha.libsonnet';
 local actions = import 'libs/actions.libsonnet';
 local semgrep = import 'libs/semgrep.libsonnet';
 
+// exported for other workflows
+local artifact_name = 'pro-ocaml-build-artifacts-release';
+
+local container = semgrep.containers.ocaml_alpine;
+
 // ----------------------------------------------------------------------------
 // The job
 // ----------------------------------------------------------------------------
 
-local job = {
-  'runs-on': 'ubuntu-latest',
-  // Switching to Ubuntu here because Alpine does not provide easily 'gh'
-  // which is needed to checkout semgrep-pro from semgrep GHA.
-  // alt: use our r2c alpine-ocaml and procedure to download gh binary tarball
+local job = container.job {
+  // The ocaml_alpine container we are using here has opam already
+  // installed, as well as an opam switch already created, and a big set of
+  // packages already installed. Thus, the
+  // 'make install-deps-ALPINE-for-semgrep-core' below is very fast and almost a
+  // noop.
```

Comment:
This explains why it's ok that we run `make install-deps-ALPINE-for-semgrep-core` but I missed why we added that step. (I assume it's related to static linking as mentioned in the PR---but I'm still a bit confused about why that's an issue)

Reply:
I see what you're saying and I'll add more comments.

Basically the docker image is a blackbox and I'm not sure if we can really expect all the required things for an alpine build to be there.